cmake/linker/ld/target_ordered.cmake

```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# See root CMakeLists.txt for description and expectations of these macros
+
+macro(toolchain_ld_ordered_options dest_var_name)
```

Comment:
Would toolchain_ld_ordered_options() -> toolchain_ld_options() and placing it to target_base.cmake be sufficient?

Regarding the order, https://sourceware.org/binutils/docs-2.32/ld/Options.html

> Some of the command-line options to ld may be specified at any point in the command line. However, options which refer to files, such as ‘-l’ or ‘-T’, cause the file to be read at the point at which the option appears in the command line, relative to the object files and other file options. Repeating non-file options with a different argument will either have no further effect, or override prior occurrences (those further to the left on the command line) of that option. Options which may be meaningfully specified more than once are noted in the descriptions below.